weaver/utils.py

```diff
@@ -2872,6 +2872,14 @@ def fetch_directory(location,                           # type: str
                                     include=include, exclude=exclude,
                                     settings=settings, **option_kwargs)
     elif location.startswith("http://") or location.startswith("https://"):
+        # Next two lines are added to match behavior of `download_files_s3` and replicate input directory name
+        # in output location
+        input_directory_name = os.path.basename(os.path.normpath(location))
+        out_dir = os.path.join(out_dir, input_directory_name)
```

Comment:
Similar question as mentioned before. 
If the location is only "host+`/`", the full hostname will be taken as the input directory name.

ex: `https://test.com/` would create a directory `{out_dir}/test.com/`

Depending on how the URL is forged, I wonder it that could create some vulnerability?

Reply:
Hmmm, there might be some vulnerability, but appart from, say `{out_dir}/../`, which would only give access to the job's directory, are there other possibilities? and besides, wouldn't it require  `https://../`, to be a valid url ? (or `https://test.com/../`)

Directory traversal would be limited because of the parsing, but I wonder what other vectors would be possible.

Reply:
A quick tests and obtained results:

<table>
   <th>Test
   <th><code>normpath</code>
   <th><code>normpath</code>+<code>basename</code>
<tr>
<td><code>https://test.com/../</code>
<td><code>https:</code>
<td><code>https:</code>
<tr>
<td><code>https://test.com/../../</code>
<td><code>.</code>
<td><code>.</code>
<tr>
<td><code>https://test.com/../../../</code>
<td><code>..</code>
<td><code>..</code>
<tr>
<td><code>https://test.com/../../../../</code>
<td><code>../..</code>
<td><code>..</code>
</table>

So depending on the amount of `../` the following join with `out_dir` could allow someone to navigate one directory higher in the hierarchy than permitted.